### ISSUE TITLE
v6.3.0 upgrade to v6.4.0 will fail when has causal-ts.available-interval config in v6.3.0

### ISSUE THREAD

**Issue by seiya-annie** [tikv/tikv#13772]
### What version of TiKV are you using?
v6.4.0-pre

### Steps to reproduce
install v6.3.0 with  causal-ts.available-interval=3s
upgrade to v6.4.0
### What did you expect?
delete config item can be ignored when upgrade
### What did happened?
```code
Error: init config failed: [172.16.5.52:2360](http://172.16.5.52:2360/): executor.ssh.execute_failed: Failed to execute command over SSH for '[liting@172.16.5.52:22](mailto:liting@172.16.5.52:22)' {ssh_stderr: unknown configuration options: causal-ts.available-interval
, ssh_stdout: , ssh_command: export LANG=C; PATH=$PATH:/bin:/sbin:/usr/bin:/usr/sbin /home/liting/deploy/tikv-2360/bin/tikv-server --config-check --config=/home/liting/deploy/tikv-2360/conf/tikv.toml --pd "" --data-dir "/data1/liting/data/tikv-2360"}, cause: Process exited with status 1: check config failed
```

**Comment by pingyu** [comment]
`causal-ts.available-interval` was renamed to `causal-ts.alloc-ahead-buffer` in https://github.com/tikv/tikv/pull/13597.

Although codes for `causal-ts.available-interval` was introduced before `v6.3.0`, but it has never appeared in any document. So we don't expect any user will use this configuration item and face issue on upgrade.

p.s. `causal-ts.alloc-ahead-buffer` will be introduced since `v6.4.0`, see [here](https://docs.pingcap.com/tidb/dev/tikv-configuration-file#alloc-ahead-buffer-new-in-v640).

**Comment by pingyu** [comment]
Won't fix.
